File: backend/models/Student.js
const mongoose = require('mongoose');

const studentSchema = mongoose.Schema({
    username: { type: String, required: true },
    email: { type: String, required: true },
    studentID: { type: String, required: true },
    dateOfBirth: { type: String, required: true },
    gender: { type: String, required: true },
    address: { type: String, required: true },
    phone: { type: String, required: true },
    major: { type: String, required: true },
    fieldName: { type: String, required: true },
    cV_URL: { type: String, required: true },
    fullName: { type: String, required: true },
    ojtStatus: Boolean
});

module.exports = mongoose.model('Student', studentSchema);

File: src/main/java/backend/model/Student.java
package backend.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "Students")
@Getter @Setter @NoArgsConstructor
public class Student {

    @Id
    @Column (name = "StudentID")
    private String id;

    @Column(name = "Username")
    private String username;

    @OneToOne
    private User user;

    @Column (name = "Major")
    private int majorId;

    @ManyToOne
    private Major major;

    @Column(name = "DateOfBirth")
    private String dob;
    @Column(name = "Gender")
    private String gender;
    @Column(name = "Address")
    private String address;
    @Column(name = "Phone")
    private String phone;
    @Column(name = "CV_URL")
    private String cvUrl;
    @Column(name = "OJTStatus")
    private boolean isInOjt;



}
